internal/status/cache.go

```diff
@@ -61,6 +74,12 @@ func (c Cache) commitProxy(pu *ProxyUpdate) {
 		return
 	}
 
+	validCond := pu.ConditionFor(ValidCondition)
+	if validCond.Reason == "" {
+		validCond.Reason = "Valid"
+		validCond.Message = "Valid HTTPProxy"
```

Comment:
This depends on behavior that might not always be true. I'd suggest that we default these when the condition is created.